problem_1/lib/multiples_of_3_5.rb

```diff
@@ -1,35 +1,18 @@
-class SumGenerator
+class MultiplesCalculator
 
 	def initialize(range)
 		@range = range
 	end
 
-	def multiples_of_3
-		@multiples_3 = []
-		(@range).each do |digit|
-			if (digit % 3) == 0
-				@multiples_3 << digit
-			end
+	def multiples_generator
+		multiples = []
+		(@range).each do |digit| 
+			(multiples << digit) if (digit % 3) == 0 || (digit % 5) == 0
 		end
-		@multiples_3
-	end
-
-	def multiples_of_5
-		@multiples_5 = []
-		(@range).each do |digit|
-			if (digit % 5) == 0
-				@multiples_5 << digit
-			end
-		end
-		@multiples_5
-	end
-
-	def sum
-		all_multiples = (multiples_of_3 + multiples_of_5).uniq
-		all_multiples.sum
+		multiples.uniq.sum
 	end
 end
 
-count = SumGenerator.new(1...1000)
+calculator = MultiplesCalculator.new(1...1000)
```

Comment:
No need to instantiate this class here it should either be done in your tests or by other classes that are using it